File: ui-frontend/src/data/company-management/establishment/establishment-product-dto.ts
import {WrappedProductDTO} from "../saleable/product/wrapped-product-dto";
import {EstablishmentSaleableDTO} from "./establishment-saleable-dto";

export interface EstablishmentProductDTO extends EstablishmentSaleableDTO {
  remainingStock: number,
  product: WrappedProductDTO
}

File: model/src/main/java/org/bandrsoftwares/celestialdiary/model/dto/establishment/EstablishmentProductDTO.java
package org.bandrsoftwares.celestialdiary.model.dto.establishment;

import org.bandrsoftwares.celestialdiary.model.dto.saleable.product.WrappedProductDTO;
import org.bandrsoftwares.celestialdiary.model.mongodb.establishment.EstablishmentProduct;

public record EstablishmentProductDTO(double customPrice, boolean usingCustomPrice, int remainingStock, WrappedProductDTO product) {

    public EstablishmentProductDTO(EstablishmentProduct establishmentProduct) {
        this(establishmentProduct.getCustomPrice(), establishmentProduct.getUsingCustomPrice(), establishmentProduct.getRemainingStock(),
             new WrappedProductDTO(establishmentProduct.getProduct()));
    }
}
